src/containers/Balance/Balance.tsx

```diff
@@ -93,17 +94,17 @@ class Balance extends React.Component<Props, State> {
     const { value: amount } = event.target
     const myBalance = this.getMyBalance()
 
-    if (!myBalance || amount.includes('.')) {
+    if (!myBalance || !new RegExp('^[0-9]+$')) {
       return
     }
 
-    const amountNumber = Number(amount)
+    const amountNumber = new BN(amount)
```

Comment:
isn't amount here already a `BN`?
Or do you need to make a copy?

Reply:
No, it is a string value which we get from the value input. 

`const { value: amount } = event.target`